### DISCUSSION TITLE
Task IDs are not unique

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [celery/celery#7308] [Q&A]
Task (UU)IDs don't seem to be unique. When the task ID is not set, Celery generates one with `uuid4()`, see:

* https://github.com/celery/celery/blob/25ca389039f667cdf940c9efe44bdefd1cf70f30/celery/canvas.py#L287
* https://github.com/celery/celery/blob/25ca389039f667cdf940c9efe44bdefd1cf70f30/celery/app/task.py#L761

No check is done to see if the task ID is already in use by a task in the result backend (which contains the `task_id` attribute for every result).

I want to make sure that task IDs are unique (yes, I know the chance of a clash is very unlikely). Am I supposed to keep track of task IDs myself and generate a unique one, or is there a more native/built-in way that I'm missing?

**Comment by thedrow** [comment]
UUIDs are [sufficiently](https://stackoverflow.com/questions/1155008/how-unique-is-uuid) unique. If you'd like to generate your own IDs, pass the id to the signature as an argument.

**Comment by fatemeh-mohseni-AI** [comment]
that is my problem too

**Comment by auvipy** [comment]
I have  open a PR to switch to uuid7... https://github.com/celery/kombu/pull/2303 may be that partially fix this issue? may be we also add some uniqueness check as well beside moving to uuid7?
